src/Microsoft.ML.StandardLearners/Standard/SdcaStatic.cs

```diff
@@ -54,17 +56,16 @@ public static partial class RegressionTrainers
             Contracts.CheckParam(!(l1Threshold < 0), nameof(l1Threshold), "Must not be negative, if specified.");
             Contracts.CheckParam(!(maxIterations < 1), nameof(maxIterations), "Must be positive if specified");
             Contracts.CheckValueOrNull(loss);
+            Contracts.CheckValueOrNull(advancedSettings);
```

Comment:
>advancedSettings [](start = 39, length = 16)

the advanced settings delegate should be inside the original ctor, no such functionality should be exclusive to pigstensions

Reply:
here I'm 26 files later, looking on this small comment and hating myself. ARE YOU HAPPY? ARE YOU????

---
In reply to: [223083700](https://github.com/dotnet/machinelearning/pull/1160#discussion_r223083700) [](ancestors = 223083700)

Reply:
Yessss!

---
In reply to: [224609957](https://github.com/dotnet/machinelearning/pull/1160#discussion_r224609957) [](ancestors = 224609957,223083700)